[BUGFIX] Add `<head>` element despite `<header>`

JakeQZ opened the pull request:
Also test around the issue.

Resolves #844.

oliverklee commented on tests/Unit/CssInlinerTest.php:
```diff
@@ -2252,6 +2252,20 @@ public function inlineCssAppliesCssToMatchingElementsAndKeepsRuleWithPseudoCompo
     }
 
+    /**
+     * #844
```
To keep the code succinct, please let's not have issue numbers in the code. Having them in the changelog and the git history should suffice.

oliverklee commented on tests/Unit/HtmlProcessor/AbstractHtmlProcessorTest.php:
```diff
@@ -175,6 +175,9 @@ public function contentWithoutHeadTagDataProvider(): array
             'doctype only' => ['<!DOCTYPE html>'],
             'body content only' => ['<p>Hello</p>'],
             'BODY element' => ['<body></body>'],
+            // #844
```
Please let's remove the issue number here as well.

JakeQZ commented:
> To keep the code succinct, please let's not have issue numbers in the code. Having them in the changelog and the git history should suffice.

OK, I've removed these.

Perhaps I am abusing the `needs unit tests` label?  I added it because the tests added went slightly beyond the specific issue.  But perhaps it is intended as a temporary label like `waiting for feedback`?

oliverklee commented:
@JakeQZ I intended the `unit tests needed`label to mean "there still are some tests missing".
